### PR TITLE
User registration

### LINKED ISSUE #4: User registration
Users should be able to register a new account.

Acceptance criteria:
- The user must have:
	* a unique email
	* a password
	* a display name
- Email verification should be required to login.
- Use gravatar initially for avatars.

### PR DESCRIPTION
**Pull request by moathdev-zz**
Fixes #4

### CODE REVIEW

File: database/migrations/2018_06_19_025133_create_verify_users_table.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+class CreateVerifyUsersTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('verify_users', function (Blueprint $table) {
```
Comment (ibrasho) [inline database/migrations/2018_06_19_025133_create_verify_users_table.php]: Can we change this to `user_verification_tokens`?

File: app/VerifyUser.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class VerifyUser extends Model
```
Comment (ibrasho) [inline app/VerifyUser.php]: `VerificationToken`?

File: app/Mail/VerifyMail.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Mail;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Mail\Mailable;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class VerifyMail extends Mailable
```
Comment (ibrasho) [inline app/Mail/VerifyMail.php]: `UserVerificationMail`?

File: app/Http/Controllers/Auth/RegisterController.php [hunk 4]
```diff
@@ -63,10 +67,33 @@ protected function validator(array $data)
...
             'password' => Hash::make($data['password']),
         ]);
+
+        $user->verifyUser()->create([
+            'token' => str_random(40)
+        ]);
+
+        Mail::to($user->email)->send(new VerifyMail($user));
+
+    ;
```
Comment (ibrasho) [inline app/Http/Controllers/Auth/RegisterController.php]: 🤔

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 5]
```diff
@@ -17,7 +17,9 @@ public function up()
             $table->increments('id');
             $table->string('name');
             $table->string('email')->unique();
+            $table->boolean('verified')->default(false);
             $table->string('password');
+            $table->softDeletes();
```
Comment (ibrasho) [inline database/migrations/2014_10_12_000000_create_users_table.php]: We need these changes in a new migration file.

File: database/migrations/2018_06_19_025133_create_verify_users_table.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateVerifyUsersTable extends Migration
```
Comment (ibrasho) [inline database/migrations/2018_06_19_025133_create_verify_users_table.php]: It'll be good if we rename the file to `create_user_verification_tokens_table.php` and the class to `CreateUserVerificationTokensTable`.

**Review by ibrasho — CHANGES_REQUESTED** [review]

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 5 above)
Comment (moathdev-zz) [reply database/migrations/2014_10_12_000000_create_users_table.php]: no need , we don't have release or something on production .

File: database/migrations/2018_06_19_025133_create_verify_users_table.php (hunk 6 above)
Comment (moathdev-zz) [reply database/migrations/2018_06_19_025133_create_verify_users_table.php]: it's duplicate i did remove it .

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 5 above)
Comment (ibrasho) [reply database/migrations/2014_10_12_000000_create_users_table.php]: We already do. 😁 

Can you at least move these changes to `database/migrations/2018_06_19_025133_create_user_verification_tokens_table.php`?

**Pull request merged**